### PR TITLE
Make Common Custom User Data Script compatible with the configuration cache.

### PR DESCRIPTION
**Pull request by runningcode**
This makes the Gradle common custom user data script compatible with the configuration cache.
I tested this in a very simple Gradle project with git installed using Gradle 6.6 and 6.7-rc-1.
I verified that the Git status custom value was correctly plublished in the build scans.

Here's a build scan I published with the configuration cache enabled using these changes: https://scans.gradle.com/s/gimsuc3veohve/custom-values#L0

### CODE REVIEW

**Review by marcphilipp — APPROVED** [review]

File: common-custom-user-data-scripts/gradle-common-custom-user-data.gradle [hunk 1]
```diff
@@ -319,7 +319,7 @@ String execAndGetStdout(String... args) {
     }
 }
 
-void addCustomLinkWithSearchTerms(def buildScan, String title, Map<String, String> search) {
+static void addCustomLinkWithSearchTerms(def buildScan, String title, Map<String, String> search) {
```
Comment (etiennestuder) [inline common-custom-user-data-scripts/gradle-common-custom-user-data.gradle:322]: How is this working without making `customValueSearchParams`  and `appendIfMissing` methods static?

File: common-custom-user-data-scripts/gradle-common-custom-user-data.gradle [hunk 2]
```diff
@@ -291,7 +291,7 @@ boolean isTravis() {
     System.getenv('TRAVIS_JOB_ID')
 }
 
-boolean isGitInstalled() {
+static boolean isGitInstalled() {
```
Comment (etiennestuder) [inline common-custom-user-data-scripts/gradle-common-custom-user-data.gradle:294]: Please make the 8 isXYZ methods above static too

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: common-custom-user-data-scripts/gradle-common-custom-user-data.gradle [hunk 3]
```diff
@@ -341,7 +341,7 @@ String appendIfMissing(String str, String suffix) {
     str.endsWith(suffix) ? str : str + suffix
 }
 
-String trimAtEnd(String str) {
+static String trimAtEnd(String str) {
     ('x' + str).trim().substring(1)
 }
```
Comment (etiennestuder) [inline common-custom-user-data-scripts/gradle-common-custom-user-data.gradle:347]: Please verify the script still works with 5.0.x, 5.6.x and 6.0.x. (not with regard to configuration caching but the general functionality)

**Comment by etiennestuder** [comment]
Once merged, you can also close #6527.

File: common-custom-user-data-scripts/gradle-common-custom-user-data.gradle (hunk 1 above)
Comment (runningcode) [reply common-custom-user-data-scripts/gradle-common-custom-user-data.gradle]: I can't say I know enough about groovy to understand this. :) I'll make them static though.

File: common-custom-user-data-scripts/gradle-common-custom-user-data.gradle (hunk 2 above)
Comment (runningcode) [reply common-custom-user-data-scripts/gradle-common-custom-user-data.gradle]: Done.

File: common-custom-user-data-scripts/gradle-common-custom-user-data.gradle (hunk 3 above)
Comment (runningcode) [reply common-custom-user-data-scripts/gradle-common-custom-user-data.gradle]: Gradle 5.6.4
https://scans.gradle.com/s/gba6luiyltwp2

Gradle 5.0
https://scans.gradle.com/s/vwtpepkn6ooiq

Gradle 6.6.1 using configuration cache
https://scans.gradle.com/s/xzqryy5s6tg4s

There were other problems with the configuration cache in 6.6.1 with this build, but the git status, git commit and git link were correctly added.

**Review by etiennestuder — APPROVED** [review]

**Comment by etiennestuder** [comment]
Thanks for the fix, @runningcode

**Pull request merged**